context.go

```diff
@@ -76,9 +76,7 @@ func getTChannelParams(ctx context.Context) *tchannelCtxParams {
 
 // NewContext returns a new root context used to make TChannel requests.
 func NewContext(timeout time.Duration) (context.Context, context.CancelFunc) {
-	return NewContextBuilder(timeout).
-		setSpan(NewRootSpan()).
```

Comment:
One use case we have is `yab` which makes a call from an outbound perspective, but then outputs the traceID (so you can find the trace later). It uses the `CurrentSpan(ctx)` after creating a context, so I wonder if every new context should include a new root span. While `yab` can be updated, I don't want to break any other uses of this pattern.